src/contrib/cluster/Akka.DistributedData/Serialization/SerializationSupport.cs

```diff
@@ -173,8 +184,14 @@ public VersionVector VersionVectorFromProto(Proto.Msg.VersionVector versionVecto
             if (entries.Count == 1)
                 return new SingleVersionVector(UniqueAddressFromProto(versionVector.Entries[0].Node),
                     versionVector.Entries[0].Version);
-            var versions = entries.ToDictionary(x => UniqueAddressFromProto(x.Node), v => v.Version);
```

Comment:
This actually calls Dictionary.GetEnumerator() in the underlying code, creating hash set bucket allocation in the background.